kombu/utils/url.py

```diff
@@ -38,6 +44,24 @@ def parse_url(url):
     # type: (str) -> Dict
     """Parse URL into mapping of components."""
     scheme, host, port, user, password, path, query = _parse_url(url)
+    if query:
+        keys = list(query.keys())
+        for key in keys:
+            if key.startswith('ssl_'):
+                if key == 'ssl_cert_reqs':
+                    if ssl_available:
+                        query[key] = getattr(ssl, query[key])
+
+                    else:
+                        query[key] = None
+                        logger.warn('Defaulting to insecure SSL behaviour.')
+
+                if 'ssl' not in query:
```

Comment:
Why is this part of the loop?

Reply:
on the first iteration of the loop `query['ssl']` may not exist. If it does not this initializes it.

Reply:
We could extract it from the loop and initialize the key if it doesn't exist.

Reply:
I didn't want to do that because then there will always be an `ssl` key in the return which could potentially overwrite kwargs passed in here: https://github.com/celery/kombu/blob/master/kombu/connection.py#L157-L171

Return from `parse_url` overrides those here: https://github.com/celery/kombu/blob/master/kombu/connection.py#L189-L193

I only want to override `ssl` if there is an ssl parameter in the URL. I forget where I saw it in Celery code but it stated query string parameters can override functions kwargs. Query strings are given the final say but if they are not there and we should initialize that in the return.